test/object-path-immutable-transaction.js

```diff
@@ -0,0 +1,581 @@
+'use strict';
+
+const tap = require('tap');
+const sinon = require('sinon');
+const immutableTransaction =
+	require('../lib/object-path-immutable-transaction');
+
+tap.test('object-path-immutable-transaction', t => {
+	t.type(immutableTransaction, 'function', 'exports a function');
+	t.throws(() => immutableTransaction(), 'requires an argument');
+
+	t.test('return value', t => {
+		t.type(immutableTransaction({}), 'object', 'object');
+		t.type(immutableTransaction({}).set, 'function', 'implements `set`');
+		t.type(immutableTransaction({}).del, 'function', 'implements `del`');
+		t.type(immutableTransaction({}).push, 'function', 'implements `push`');
+		t.type(immutableTransaction({}).assign, 'function', 'implements `assign`');
+
+		t.end();
+	});
+
+	t.test('set', t => {
+		t.test('does nothing if not passed enough arguments', t => {
```

Comment:
I don't need to nitpick if you wanna get this in, but I'm curious about the motivation for that design decision.